### PR TITLE
Support mac through libusdt

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by ghost** [comment]
Can you have a repo be its own builder like that? I've never seen that, but i guess it makes sense!

**Comment by vchuravy** [comment]
Yeah you just need a place to dump the binary results xD

**Comment by vchuravy** [comment]
Actually we might be able to be sneaky and compile to a LLVM bc file.isntead of a shared library.

**Comment by NHDaly** [comment]
> Actually we might be able to be sneaky and compile to a LLVM bc file.isntead of a shared library.

Makes sense. Why is that better?

**Comment by vchuravy** [comment]
> Makes sense. Why is that better?

We might be able to get rid of a `ccall` by doing that and replace it either by inlining of having a normal call, but that is an optimisation for later.

File: src/libusdt.jl [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    # int usdt_provider_disable(usdt_provider_t *provider);
+    ccall((:usdt_provider_disable, libusdt), Nothing, (Ptr{Provider},), provider)
+end
+function provider_free(provider)
+    # void usdt_provider_free(usdt_provider_t *provider);
+    ccall((:usdt_provider_free, libusdt), Nothing, (Ptr{Provider},), provider)
+end
+
+# TODO: varargs...?
+#function error(provider, error, args...)
```
Comment (vchuravy) [inline src/libusdt.jl]:
```suggestion
function error(provider, error, args...)
  ccall((:usdt_error, libusdt), Nothing, (Ptr{Provider}, Cint, Cstring...), provider, error, args...)
end
```

**Comment by NHDaly** [comment]
Is this PR still needed? Does UProbes support mac already now?

**Comment by vchuravy** [comment]
I never had the opportunity to test on a Mac, and I don't recall why I thought this would be necessary.

**Comment by NHDaly** [comment]
Closing for now; we can always open again if needed. Thanks

**Pull request closed without merging**